xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1056 @@
+from xdsl.printer import Printer
+from xdsl.traits import IsTerminator, SymbolOpInterface, SymbolTable
+
+i8 = IntegerType(8)
+
+CASCADE_SIZE = 384
+
+LOCK_ACQUIRE = 1
+LOCK_RELEASE = 0
+
+PRODUCE_PORT = 0
+CONSUME_PORT = 1
+
+BLOCKING = 1
+NONBLOCKING = 0
+
+MM2S = 0
+S2MM = 1
+
+i1 = IntegerType(1)
+
+
+# Hexadecimal values must be formatted with a lowercase '0x' preceeding the number
+def format_hex(value: int):
+    return hex(value).upper().replace("X", "x")
+
+
+@irdl_attr_definition
+class WireBundleAttr(Data[str]):
+    name = "AIE.wire_bundle"
+
+    @classmethod
+    def parse_parameter(cls, parser: AttrParser) -> str:
+        return parser.parse_str_literal()
+
+    def print_parameter(self, printer: Printer) -> None:
+        printer.print_string(f'"{self.data}"')
+
+
+@irdl_attr_definition
+class ObjectFIFO(ParametrizedAttribute):
+    name = "AIE.objectFifo"
+
+    buffer: ParameterDef[memref.MemRefType]
+
+    @staticmethod
+    def from_element_type_and_shape(
+        referenced_type: _MemRefTypeElement,
+        shape: Iterable[int | AnyIntegerAttr],
+        layout: Attribute = NoneAttr(),
+        memory_space: Attribute = NoneAttr(),
+    ) -> ObjectFIFO[_MemRefTypeElement]:
+        return ObjectFIFO(
+            [
+                memref.MemRefType.from_element_type_and_shape(
+                    referenced_type, shape, layout, memory_space
+                )
+            ]
+        )
+
+
+@irdl_attr_definition
+class ObjectFIFOSubview(ParametrizedAttribute):
+    name = "AIE.objectFifoSubview"
+
+    buffer: ParameterDef[memref.MemRefType]
+
+    @staticmethod
+    def from_element_type_and_shape(
+        object_fifo: ObjectFIFO,
+        shape: Iterable[int | AnyIntegerAttr],
+    ) -> ObjectFIFOSubview[_MemRefTypeElement]:
+        return ObjectFIFOSubview(
+            [
+                memref.MemRefType.from_element_type_and_shape(
+                    object_fifo.buffer.element_type,
+                    shape,
+                    object_fifo.buffer.layout,
+                    object_fifo.buffer.memory_space,
+                )
+            ]
+        )
+
+
+@irdl_op_definition
+class AMSelOp(IRDLOperation):
+    name = "AIE.amsel"
+    arbiterID: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+    msel: IntegerAttr[i32] = attr_def(IntegerAttr[i32])
+    result: OpResult = result_def(IndexType())
+
+    def __init__(self, arbiterID: IntegerAttr[i32], msel: IntegerAttr[i32]):
+        super().__init__(
+            attributes={"arbiterID": arbiterID, "msel": msel},
+            result_types=[IndexType()],
+        )
+
+
+@irdl_op_definition
+class BufferOp(IRDLOperation):
+    name = "AIE.buffer"
+    tile: Operand = operand_def(IndexType())
+    shape: ArrayAttr[i32] = attr_def(ArrayAttr[i32])
+    element_type: AnyAttr = attr_def(AnyAttr())
+    buffer: OpResult = result_def(memref.MemRefType)
+    sym_name: StringAttr = attr_def(StringAttr)
+
+    def __init__(
+        self,
+        tile: IndexType(),
```

Comment:
Would it make sense for this to be 

```suggestion
        tile: SSAValue | Operation,
```

Reply:
Similarly in other places.